### ISSUE TITLE
Struggling to update offline setup from 0.1.0 to 0.2.3

### ISSUE THREAD

**Issue by epugh** [jupyterlite/jupyterlite#1311]
## Description

I have a public opensource project for Quepid that wraps up our Jupterlite setup.  I am attempting to upgrade from 0.1.0 to 0.2.3 and the upgrade path hasn't been as clear as I was hoping.   

One wrinkle is that we require the offline mode.  If I do an online mode, it works, but offline it doesn't, and I suspect it's cause I'm not using the right pyodide?

## Reproduce

Here is what I do:

```
git@github.com:o19s/quepid-jupyterlite.git
git checkout bump_jupyterlite_0_2x
docker run -it --rm -e TARGET_DIR=/dist -v "$(pwd)":/dist $(docker build -q .)
untar -xvf jupyter-lite-build.tgz
python3 -m http.server 8000 --directory ./notebooks
```

Open it up on http://localhost:8000 and I have kernals, but when I go to load them I get an error about missing files and `wasm not instantiated`.

## Expected behavior

It runs offline!

## Context

- JupyterLite version: 0.2.3
- Operating System and version:  OSX 
- Browser and version: Chrome

<details><summary>Browser Output</summary>
<img width="643" alt="image" src="https://github.com/jupyterlite/jupyterlite/assets/22395/010455e3-cc4d-4e74-8d3a-fc10d077e774">

**Comment by jtpio** [comment]
Thanks @epugh for the report.

The downloading of a specific Pyodide version that can be used offline is now handled in the `jupyterlite-pyodide-kernel` addon: https://github.com/jupyterlite/pyodide-kernel

Normally the `--pyodide` CLI option should indeed still be supported: https://jupyterlite-pyodide-kernel.readthedocs.io/en/latest/?badge=latest#usage

**Comment by jtpio** [comment]
Ah according to the screenshot in https://github.com/o19s/quepid-jupyterlite/pull/3, it looks like this might be related to the mimetype of the wasm file.

Maybe you could try starting the server with `jupyter lite serve` to see if that may help? (this command should be able to server `.wasm` files with the proper mime type)

**Comment by epugh** [comment]
I made some progress...    I did not have luck with using `jupyter lite serve`, I think maybe not quite getting the build process to work...  However, I did create a `server.py` (and checked in on the PR) that includes the `wasm` mimetype, which gets me further.

Running `python server` in the root and opening http://localhost:8080/notebooks/lab/index.html loads up the homepage with kernals listed!

Creating a python notebook does run into:

<img width="650" alt="image" src="https://github.com/jupyterlite/jupyterlite/assets/22395/7bdbe09d-6809-428a-8763-ec0ef9a416d3">

Do I need to add more modules to my piplite listing?

**Comment by jtpio** [comment]
Not sure if it may be an issue with the offline Pyodide feature, or related to the recent update to Pyodide 0.25.0 released in `jupyterlite-pyodide-kernel==0.2.2`.

Would you be able to to try with `jupyterlite-pyodide-kernel==0.2.1`, which uses a previous version of Pyodide?

**Comment by epugh** [comment]
okay, by going with `jupyterlite-pyodide-kernel==0.2.1` I now load some modules, and then it errors on out `BaseComm`?

<img width="415" alt="image" src="https://github.com/jupyterlite/jupyterlite/assets/22395/2c873bfc-3e57-4971-becf-3de21cbd9461">

I did notice that by removing in my docker command the `--pyodide` command that my tgz file is much smaller, I suspect because I am not packaging it for offline mode?

**Comment by jtpio** [comment]
> I did notice that by removing in my docker command the `--pyodide` command that my tgz file is much smaller, I suspect because I am not packaging it for offline mode?

Yes, the full pyodide distribution can be quite large when downloaded locally.

> okay, by going with `jupyterlite-pyodide-kernel==0.2.1` I now load some modules, and then it errors on out `BaseComm`?

Ah, this may be an issue with a recent ipywidgets release.

As an alternative you may want to consider using the `xeus-python` kernel as an alternative to Pyodide: https://jupyterlite.readthedocs.io/en/latest/howto/xeus-python/preinstalled_packages.html. It makes pre-installing packages a bit easier and removes the need for using `piplite`. You may have to convert the list of packages from your `jupyter_lite_config.json` to an `environment.yml`. But the packages would be available in the built site and available for offline by default.
